flamingo/scripts/bh_jet_energies.py

```diff
@@ -0,0 +1,118 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from swiftsimio import load
+
+from unyt import mh, cm, Gyr
+from matplotlib.colors import LogNorm
+from matplotlib.animation import FuncAnimation
+
+# Set the limits of the figure.
+mass_bounds = [1e5,3e10] 
+value_bounds = [1e58,1e63]
+bins=25
+def_value = -1.
+
+def get_data(filename):
+
+    data = load(filename)
+
+    masses = data.black_holes.subgrid_masses.to("Msun")
+    jet_energies = np.array(data.black_holes.injected_jet_energies.value)
+    values = 2*1e53*jet_energies
+    
+    if np.size(values)==0:
```

Comment:
This was for a special case where jet-related fields don't exist, if the run is using e.g. the fiducial model. In this case a pipeline comparison won't work. This condition made sure that the plot was created anyways.

In the general case it might not be necessary to include this, since that type of comparison might not happen often. Should I remove it?

Reply:
I don't understand how this would run if jet-related fields don't exist. The script should then produce an error on the line where you try to access those variables, i.e.
```
    jet_energies = np.array(data.black_holes.injected_jet_energies.value)
```

The proper way to deal with this is by using a `try..except` block, which will catch the error.

Reply:
I have now introduced `try` and `except` wherever jet or spin-related quantities pop up.

Reply:
Thanks, but I am not sure your solution actually works. Can you test this on a snapshot that is missing the spin-related quantities? `unyt_array`s and `numpy` arrays usually don't mix well, so it is probably safer to use a proper `unyt_array` as the fallback instead of a plain numpy array. That way you can also be sure that the dtype and units are consistent in both cases.

Reply:
Thanks, that's a good point. I define these as `unyt_arrays`, in the way that you suggested below.